pkg/operator/helpers/helpers_test.go

```diff
@@ -1572,15 +1573,15 @@ func TestConvertToFeatureGateFlags(t *testing.T) {
 		{
 			name:         "unset",
 			features:     []operatorapiv1.FeatureGate{},
-			desiredFlags: []string{},
+			desiredFlags: []string{"--feature-gates=ClusterClaim=true", "--feature-gates=AddonManagement=true"},
```

Comment:
if it is unset, should we keep the desireFlags unset?

Reply:
If the user's provided featureGate is nil, the default features will be enabled automatically. So, the desiredFlags set to the default featureGate

Reply:
I am thinking of the case: if we set the default value in the flags explicitly(TestFeature=false), once we change the default value in the future(TestFeature=true), will the TestFeature be changed(false to true)? This is going to be hard to decide because we do not know whether the old value(TestFeature=false) is set by default or users. what is our expectation in this case?

Reply:
If the user explicitly sets a value for a feature gate, we’re always taking that as the defined value. If the user doesn’t set any value for feature gates where Default=True, then we’re enabling those feature gates.

Reply:
If the user doesn’t set any value for feature gates(TestFeature) where Default=False, so this PR will set the flag TestFeature=False right? If we change the TestFeature Default=True in the next release, when upgrading from this release to the next release, how do we know if the old flag `TestFeature=False` is set by users on purpose or set by us by default? I am asking because I think if the user intentionally disabled a feature by setting TestFeature=False, we should not enable it when upgrading.

Reply:
Got it, updated. PTAL @zhujian7